src/readbin/compare.ml

```diff
@@ -0,0 +1,133 @@
+open Core_kernel.Std
+open Cmdliner
+open Bap.Std
+open Bap_plugins.Std
+
+exception No_unstripped_file
+
+let print_source = function
+  | `bw -> print_endline "bw"
+  | _ -> print_endline "others"
+
+let source = [
+  "BW", `bw;
+  "SymTbl", `symtbl;
+  "User", `user;
+  "Ida", `ida;
+]
+
+let source_gt = [
+  "User", `user;
+  "SymTbl", `symtbl
+]
+
+let tool : _ Term.t =
+  let doc = "The source of the function start result that we are going to
+  evaluate. " ^ (Arg.doc_alts_enum source) in
+  Arg.(value & pos 1 (enum source) `bw & info [] ~docv:"tool" ~doc)
+
+let gt : _ Term.t =
```

Comment:
It is not the source of ground truth, this is a type of the source. And again, writing this `-u required` is bad. (And what is "u", anyway?) You should capture this by the grammar of your interface. For example, if user provided symbol file (`--syms`) this clearly defines, that he is going to evaluate your tool vs. the provided set of symbols. So what is the need for having this redundant `gt` command line parameter.